plugins/modules/digital_ocean_droplet.py

```diff
@@ -313,20 +324,31 @@ def ensure_power_on(self, droplet_id):
             time.sleep(min(2, end_time - time.time()))
         self.module.fail_json(msg='Wait for droplet powering on timeout')
 
+    def ensure_power_off(self, droplet_id):
+        response = self.rest.post('droplets/{0}/actions'.format(droplet_id), data={'type': 'power_off'})
+        end_time = time.time() + self.wait_timeout
+        while time.time() < end_time:
+            response = self.rest.get('droplets/{0}'.format(droplet_id))
+            json_data = response.json
+            if json_data['droplet']['status'] == 'off':
+                return json_data
+            time.sleep(min(2, end_time - time.time()))
+        self.module.fail_json(msg='Wait for droplet powering off timeout')
+
 
 def core(module):
     state = module.params.pop('state')
     droplet = DODroplet(module)
-    if state == 'present':
-        droplet.create()
+    if state == 'present' or state == 'active' or state == 'inactive':  # These states imply that we have, or create, the droplet
```

Comment:
Grammar nit: the verb tense in this comment is confusing to me and I'm not sure exactly what it's trying to say.